lib/plugins/helper/toc.ts

```diff
@@ -102,4 +104,27 @@ function tocHelper(str: string, options: Options = {}) {
   return result;
 }
 
+function getAndTruncateTocObj(str: string, options: {min_depth: number, max_depth: number}, max_items: number) {
+  let data = tocObj(str, { min_depth: options.min_depth, max_depth: options.max_depth });
+
+  if (data.length === 0) {
+    return data;
+  }
+  if (max_items < 1 || max_items === Infinity) {
+    return data;
+  }
+
+  const levels = data.map(item => item.level);
+  const min = Math.min(...levels);
+  const max = Math.max(...levels);
+
+  for (let currentLevel = max; data.length > max_items && currentLevel > min; currentLevel--) {
+    data = data.filter(item => item.level < currentLevel);
+  }
```

Comment:
This might be inefficient (iterating data over and over again). But I don't have any better idea for now.